packages/profiling-node/src/integration.ts

```diff
@@ -24,8 +31,296 @@ function addToProfileQueue(profile: RawThreadCpuProfile): void {
+    }
+
+    const scope = getCurrentScope();
+    const isolationScope = getIsolationScope();
+
+    // Extract the trace_id from the current scope and assign start_timestamp as well as the new chunk_id.
+    this._chunkId = uuid4();
+    this._chunkStartTraceID = {
+      ...isolationScope.getPropagationContext(),
+      ...scope.getPropagationContext(),
+    }.traceId;
+    this._chunkStartTimestampMS = timestampInSeconds();
+
+    CpuProfilerBindings.startProfiling(this._chunkId);
+    DEBUG_BUILD && logger.log(`[Profiling] starting profiling chunk: ${this._chunkId}`);
+
+    this._chunkTimer = global.setTimeout(() => {
+      DEBUG_BUILD && logger.log(`[Profiling] Stopping profiling chunk: ${this._chunkId}`);
+      this.stop();
+      DEBUG_BUILD && logger.log('[Profiling] Starting new profiling chunk.');
+      setImmediate(this.start.bind(this));
+    }, this._chunkIntervalMS);
+
+    // Unref timeout so it doesn't keep the process alive.
+    this._chunkTimer.unref();
+  }
+
+  /**
+   * Stops the current chunk and flushes the profile to Sentry.
+   * @returns void
+   */
+  public stop(): void {
+    if (!this._client) {
+      DEBUG_BUILD &&
+        logger.log('[Profiling] Failed to collect profile, sentry client was never attached to the profiler.');
+      return;
+    }
+    if (!this._chunkId) {
+      DEBUG_BUILD &&
+        logger.log(`[Profiling] Failed to collect profile for: ${this._chunkId}, the chunk_id is missing.`);
+      return;
+    }
+    if (this._chunkTimer) {
+      global.clearTimeout(this._chunkTimer);
+    }
+    DEBUG_BUILD && logger.log(`[Profiling] Stopping profiling chunk: ${this._chunkId}`);
+
+    const profile = CpuProfilerBindings.stopProfiling(this._chunkId, 1);
+    if (!profile || !this._chunkStartTimestampMS) {
+      DEBUG_BUILD && logger.log(`[Profiling] _chunkiledStartTraceID to collect profile for: ${this._chunkId}`);
+      return;
+    }
+    if (profile) {
+      DEBUG_BUILD && logger.log(`[Profiling] Sending profile chunk ${this._chunkId}.`);
+    }
+
+    DEBUG_BUILD && logger.log(`[Profiling] Profile chunk ${this._chunkId} sent to Sentry.`);
+    const chunk = createProfilingChunkEvent(
+      this._chunkStartTimestampMS,
+      this._client,
+      this._client.getOptions(),
+      profile,
+      {
+        chunk_id: this._chunkId,
+        trace_id: this._chunkStartTraceID,
+        profiler_id: this._profilerId,
+      },
+    );
+
+    if (!chunk) {
+      DEBUG_BUILD && logger.log(`[Profiling] Failed to create profile chunk for: ${this._chunkId}`);
+      this._reset();
+      return;
+    }
+
+    this._flush(chunk);
+    // Depending on the profile and stack sizes, stopping the profile and converting
+    // the format may negatively impact the performance of the application. To avoid
+    // blocking for too long, enqueue the next chunk start inside the next macrotask.
+    // clear current chunk
+    this._reset();
+  }
+
+  private _flush(chunk: ProfileChunk): void {
```

Comment:
l: I think we should use [`FinalizationRegistry`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/FinalizationRegistry) to call this when profiler gets GC'd, this ensures we have more chances to flush out data.

Benifits of finally allowing Node 14+ 😭

Reply:
Hmm, I dont think that works. The profiler should exist throughout the lifecycle of a process and we want to continuously "stream" these chunks to the server

Reply:
yeah I realize the references make this not work - we can ignore this.